File: src/main/java/net/atos/zac/app/identity/model/RESTUser.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.identity.model;

import jakarta.validation.constraints.NotNull;

public class RESTUser {

    @NotNull public String id;

    public String naam;
}

File: src/main/kotlin/net/atos/zac/app/identity/model/RestUser.kt
/*
 * SPDX-FileCopyrightText: 2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.identity.model

import nl.info.zac.identity.model.User
import nl.info.zac.identity.model.getFullName
import nl.info.zac.util.AllOpen
import nl.info.zac.util.NoArgConstructor

@NoArgConstructor
@AllOpen
data class RestUser(
    var id: String,

    var naam: String
)

fun User.toRestUser() =
    RestUser(
        this.id,
        this.getFullName()
    )

fun List<User>.toRestUsers(): List<RestUser> =
    this.map { it.toRestUser() }
